lib/browser/modules/selection-state.js

```diff
@@ -124,15 +124,26 @@ selectionState.service('SelectionStateService', function() {
   };
 
   /**
-   * @summary Clear all selections
+   * @summary Clear selections based on parameter passed
    * @function
    * @public
    *
+   * @param {Object} options - options
+   *
    * @example
    * SelectionStateService.clear();
+   *
+   * @example
+   * SelectionStateService.clear(options);
    */
-  this.clear = function() {
-    selection = {};
+  this.clear = function(options) {
+    if (typeof options === 'object'){
+      if (options.preserveImage){
+        delete selection.drive;
```

Comment:
Instead of saying "if preserve image, delete the drive", which sounds weird at first, but makes sense since `drive` is the only other possible selection, you can do something like this:

```js
if (options && options.preserveImage) {
    selection = _.pick(selection, 'image');
} else {
    selection = {};
}
```

This has the added benefit that if we decide to store something else than a drive or an image as a selection, we won't need to remember to modify this function to delete the new property.

You can check `_.pick` documentation here: https://lodash.com/docs#pick.